spring-native/src/main/java/io/opentelemetry/example/graal/OtlpExporterConfiguration.java

```diff
@@ -0,0 +1,84 @@
+package io.opentelemetry.example.graal;
+
+import static io.opentelemetry.exporter.otlp.internal.OtlpConfigUtil.DATA_TYPE_TRACES;
+
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporterBuilder;
+import io.opentelemetry.exporter.otlp.internal.OtlpConfigUtil;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.AutoConfigureListener;
+import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterProvider;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configuration for the OTLP exporter. This configuration will replace the default OTLP exporter,
+ * and will add a custom header to the requests.
+ */
+@Configuration
+public class OtlpExporterConfiguration {
+
+  @Bean
+  ConfigurableSpanExporterProvider otlpSpanExporterProvider() {
+    return new OtlpSpanExporterProvider();
+  }
+
+  private static class OtlpSpanExporterProvider
+      implements ConfigurableSpanExporterProvider, AutoConfigureListener {
+
+    private final AtomicReference<MeterProvider> meterProviderRef =
```

Comment:
Could you explain the concurrency issue?

Reply:
No concurrency issue - this is just copied from the SDK auto config.